frontend/src/pages/RepoDetails.tsx

```diff
@@ -0,0 +1,104 @@
+import { useQuery } from '@apollo/client'
+import {
+  faCodeFork,
+  faExclamationCircle,
+  faHistory,
+  faStar,
+  faUsers,
+} from '@fortawesome/free-solid-svg-icons'
+import { GET_REPOSITORY_DATA } from 'api/queries/repositoryQueries'
+import { toast } from 'hooks/useToast'
+import { useEffect, useState } from 'react'
+import { useParams } from 'react-router-dom'
+import { formatDate } from 'utils/dateFormatter'
+import { ErrorDisplay } from 'wrappers/ErrorWrapper'
+import DetailsCard from 'components/CardDetailsPage'
+import LoadingSpinner from 'components/LoadingSpinner'
+
+const RepoDetailsPage = () => {
+  const { projectKey, repositoryKey } = useParams()
+  const [repository, setRepository] = useState(null)
+
+  const {
+    data,
+    loading: isGraphQlDataLoading,
+    error: graphQLRequestError,
+  } = useQuery(GET_REPOSITORY_DATA, {
+    variables: { projectKey: `www-project-${projectKey}`, repoKey: repositoryKey },
+  })
+
+  useEffect(() => {
+    if (data) {
+      setRepository(data?.project?.repositories[0])
+    }
+    if (graphQLRequestError) {
+      toast({
+        variant: 'destructive',
+        title: 'GraphQL Request Failed',
+        description: 'Unable to complete the requested operation.',
+      })
+    }
+  }, [data, graphQLRequestError, projectKey])
+
+  if (isGraphQlDataLoading)
+    return (
+      <div className="flex min-h-[60vh] items-center justify-center">
+        <LoadingSpinner imageUrl="/img/owasp_icon_white_sm.png" />
+      </div>
+    )
+
+  if (!isGraphQlDataLoading && !repository)
+    return (
+      <ErrorDisplay
+        statusCode={404}
+        title="Repository not found"
+        message="Sorry, the Repository you're looking for doesn't exist"
+      />
+    )
+
+  const repositoryDetails = [
+    {
+      label: 'Last Updated',
+      value: formatDate(repository.updatedAt),
+    },
+    {
+      label: 'License',
```

Comment:
The Frontend looks good to me! 👍🏼 
My only concern was with the `License` field when we don't have anything. Right now it's just an empty line, but what do you think about saying something like `License: unknown`?